BraveWallet/Crypto/NetworkSelectionView.swift

```diff
@@ -0,0 +1,320 @@
+          set: { if !$0 { store.detailNetwork = nil } }
+        ),
+        destination: {
+          if let detailNetwork = store.detailNetwork {
+            NetworkSelectionDetailView(
+              networks: detailNetwork.subNetworks,
+              selectedNetwork: networkStore.selectedChain,
+              selectedNetworkHandler: { network in
+                selectNetwork(network)
+              }
+            )
+          }
+        },
+        label: { EmptyView() }
+      )
+    )
+    .background(
+      Color.clear
+        .alert(
+          isPresented: $store.isPresentingNextNetworkAlert
+        ) {
+          Alert(
+            title: Text(String.localizedStringWithFormat(Strings.Wallet.createAccountAlertTitle, store.nextNetwork?.shortChainName ?? "")),
+            message: Text(Strings.Wallet.createAccountAlertMessage),
+            primaryButton: .default(Text(Strings.yes), action: {
+              store.handleCreateAccountAlertResponse(shouldCreateAccount: true)
+            }),
+            secondaryButton: .cancel(Text(Strings.no), action: {
+              store.handleCreateAccountAlertResponse(shouldCreateAccount: false)
+            })
+          )
+        }
+    )
+    .background(
+      Color.clear
+        .sheet(
+          isPresented: $store.isPresentingAddAccount
+        ) {
+          NavigationView {
+            AddAccountView(keyringStore: keyringStore, preSelectedCoin: store.nextNetwork?.coin)
+          }
+          .onDisappear {
+            Task { @MainActor in
+              await store.handleDismissAddAccount()
+            }
+          }
+        }
+    )
+  }
+  
+  private func selectNetwork(_ network: BraveWallet.NetworkInfo) {
+    Task { @MainActor in
+      if await store.selectNetwork(network: network) {
+        presentationMode.dismiss()
+      }
+    }
+  }
+}
+
+/*
+#if DEBUG
+struct NetworkSelectionView_Previews: PreviewProvider {
+  static var previews: some View {
+    NavigationView {
+      NetworkSelectionView(
+        networkStore: .previewStore
+      )
+    }
+  }
+#endif
+*/
+
+private struct NetworkRowView: View {
+
+  var presentation: NetworkSelectionStore.NetworkPresentation
+  var selectedNetwork: BraveWallet.NetworkInfo
+  var detailTappedHandler: (() -> Void)?
+
+  @ScaledMetric private var length: CGFloat = 30
+  
+  init(
+    presentation: NetworkSelectionStore.NetworkPresentation,
+    selectedNetwork: BraveWallet.NetworkInfo,
+    detailTappedHandler: (() -> Void)? = nil
+  ) {
+    self.presentation = presentation
+    self.selectedNetwork = selectedNetwork
+    self.detailTappedHandler = detailTappedHandler
+  }
+
+  @ViewBuilder private var checkmark: some View {
+    Group {
+      if presentation.network == selectedNetwork || presentation.subNetworks.contains(selectedNetwork) {
+        Image(systemName: "checkmark")
+      } else {
+        Image(systemName: "checkmark").hidden()
+      }
+    }
+    .foregroundColor(Color(.braveLabel))
```

Comment:
maybe we can add a accessibilityLabel for this checkmark

Reply:
Fixed up accessibility in [dc060ad](https://github.com/brave/brave-ios/pull/5706/commits/dc060add6cc1f5a14227f46e77af0ff3cd41cae7).